### ISSUE TITLE
Add cross-correlation image registration

### ISSUE THREAD

**Issue by gau-nernst** [kornia/kornia#2247]
## 🚀 Feature

Add image registration method by cross-correlation as described in https://opg.optica.org/viewmedia.cfm?r=1&rwjcode=ol&uri=ol-33-2-156

Basic ideas: the peak of cross-correlation map is the amount of translation required. Cross-correlation can be computed efficiently in Fourier domain. Sub-pixel registration can be done by upsampling the Fourier map.

Official Matlab code: https://www.mathworks.com/matlabcentral/fileexchange/18401-efficient-subpixel-image-registration-by-cross-correlation

Scikit-image implementation: https://scikit-image.org/docs/stable/api/skimage.registration.html#skimage.registration.phase_cross_correlation

## Pitch

PyTorch already has functions for FFT since 1.8: https://pytorch.org/blog/the-torch.fft-module-accelerated-fast-fourier-transforms-with-autograd-in-pyTorch/

## Alternatives

This algorithm only models translation. I'm not sure if this would be a meaningful addition to Kornia. The output is an index (argmax), so it's not differentiable.

## Additional context

______________________________________________________________________

#### Consider also to contribute to Kornia universe projects :)

<sub>

- [**Tutorials**](https://github.com/kornia/tutorials): our repository containing the tutorials.

</sub>

**Comment by ducha-aiki** [comment]
We have this one, it is general, although it is not a closed-form solution, but rely on Pytorch optimizer 
https://kornia.readthedocs.io/en/latest/geometry.transform.html#kornia.geometry.transform.image_registrator.ImageRegistrator

**Comment by gau-nernst** [comment]
Yes, I saw that one. It's pretty cool to use gradient descent for registration. That's why I'm also not sure if such a closed-form solution for a translation-only model would be a meaningful addition to kornia.

Another note. Would having a correlation-based loss meaningful for image registration? (instead of using L1 for matching loss)

**Comment by edgarriba** [comment]
> Would having a correlation-based loss meaningful for image registration? (instead of using L1 for matching loss)

that more useful i think

**Comment by ducha-aiki** [comment]
Well, it would be useful, also it would help us to finally implement KCF tracker, e.g. see here
https://cw.fel.cvut.cz/wiki/courses/mpv/labs/4b_tracking/start

**Comment by gau-nernst** [comment]
I will close this issue and open a new one for correlation loss
